internal/auth/authenticator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/internal/boxcli/usererr"
+	"go.jetpack.io/devbox/internal/cuecfg"
+	"go.jetpack.io/devbox/internal/envir"
+)
+
+// Authenticator performs various auth0 login flows to authenticate users.
+type Authenticator struct {
+	ClientID string
+	Domain   string
+	Scope    string
+	Audience string
+}
+
+// NewAuthenticator creates an authenticator that uses the auth0 production
+// tenancy.
+func NewAuthenticator() *Authenticator {
+	return &Authenticator{
+		ClientID: envir.GetValueOrDefault(
+			"DEVBOX_AUTH_CLIENT_ID",
+			"5PusB4fMm6BQ8WbTFObkTI0JUDi9ahPC",
+		),
+		Domain: envir.GetValueOrDefault(
+			"DEVBOX_AUTH_DOMAIN",
+			"auth.jetpack.io",
```

Comment:
Does that mean if we ship this cli version, we'd have to use `auth.jetpack.io` even if we choose a different domain later on?

Reply:
If we ever change auth domains, we would need to maintain this one as long as users are on this version of the CLI.

Reply:
This code won't follow redirects?